### PR TITLE
standard suite for HF, MP2.5, MP3, LCCD grad

### PR DESCRIPTION
**Pull request by loriab**
## Description
Add reference data for HF gradients, MP3 energies & gradients, and LCCD gradients. Expand contract machinery for HF, MP2.5, MP3, LCCD. Harvest MP3 from Cfour

Also converted the psi4 (non-nightly) conda-env into a "minimum" testing environment.

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: qcengine/programs/tests/standard_suite_ref.py [hunk 1]
```diff
@@ -3270,6 +4050,53 @@
                         calc["data"]["SCS-MP2 CORRELATION ENERGY"] + calc["data"]["HF TOTAL ENERGY"]
                     )
 
+        if "MP3 CORRELATION ENERGY" in calc["data"]:
```
Comment (dotsdl) [inline qcengine/programs/tests/standard_suite_ref.py:4053]: It's not clear to me why there's nesting in this sequence of conditions. Can some of the nesting be done without?

File: qcengine/programs/tests/standard_suite_ref.py [hunk 2]
```diff
@@ -768,12 +1001,15 @@
...
         "data": {
             "HF TOTAL ENERGY": _scf_nh2_adz_pk_rohf,
+            "HF TOTAL GRADIENT": _grad_scf_nh2_adz_pk_rohf,
             "MP2 CORRELATION ENERGY": -0.15702660833165538,
             "MP2 SINGLES ENERGY": -0.0028059971624814647,
             "MP2 SAME-SPIN CORRELATION ENERGY": -0.03466304269235235,
             "MP2 TOTAL GRADIENT": np.array(
                 [0.0, 0.0, 0.026398091851, 0.0, 0.014012163884, -0.013199045925, 0.0, -0.014012163884, -0.013199045925,]
             ).reshape((-1, 3)),
+            "MP3 CORRELATION ENERGY": -0.173406218947,  # cfour
```
Comment (dotsdl) [inline qcengine/programs/tests/standard_suite_ref.py:1011]: We don't currently have CI in place for cfour. Are the values derived from cfour going to be exercised at all by our current crop of tests?

File: qcengine/programs/cfour/harvester.py [hunk 3]
```diff
@@ -227,6 +229,8 @@ def harvest_outfile_pass(outtext):
         psivar["MP3 TOTAL ENERGY"] = mobj.group(8)
         psivar["MP2.5 CORRELATION ENERGY"] = dmp2 + Decimal("0.500000000000") * dmp3
         psivar["MP2.5 TOTAL ENERGY"] = psivar["MP2.5 CORRELATION ENERGY"] + psivar["SCF TOTAL ENERGY"]
+        psivar["MP3 SINGLES ENERGY"] = Decimal(mobj.group(1)) + Decimal(mobj.group(5))
```
Comment (dotsdl) [inline qcengine/programs/cfour/harvester.py:232]: To my knowledge, we don't have CI or tests exercising these codepaths. Do we accept this for now?

Comment (loriab) [reply qcengine/programs/cfour/harvester.py:232]: yeah, I've tested them over at qcdb. at that was only backup testing for psi4 (which only has one open-shell implementation of mp3). afraid testing is trickle-down to qcng from other projects.

File: qcengine/programs/tests/standard_suite_ref.py (hunk 2 above)
Comment (loriab) [reply qcengine/programs/tests/standard_suite_ref.py:1011]: current crop, no. I'll expand the stdsuite testing here sometime to at least psi4 and cfour. but it isn't dead code, as it'll be used by https://github.com/psi4/psi4/pull/1971

File: qcengine/programs/tests/standard_suite_ref.py (hunk 1 above)
Comment (loriab) [reply qcengine/programs/tests/standard_suite_ref.py:4053]: Many implementations are incomplete wrt subsidiary figures (e.g., singles energy). So unless one can collect a complete set of ref values for all r/u/rohf, conv/df/cd combinations, it's safer to do the nesting.

Comment (dotsdl) [reply qcengine/programs/tests/standard_suite_ref.py:4053]: Ah, thank you! This clarifies the intent for me. I'm not familiar enough with the interdependence of these figures to reason about what the nesting means here.

File: qcengine/programs/tests/standard_suite_ref.py (hunk 2 above)
Comment (dotsdl) [reply qcengine/programs/tests/standard_suite_ref.py:1011]: Ah cool, thank you! This sounds good to me; I don't want it to block merge of this substantial work!

File: qcengine/programs/cfour/harvester.py (hunk 3 above)
Comment (dotsdl) [reply qcengine/programs/cfour/harvester.py:232]: Understood, thank you! This clarifies it for me as well. Appreciated!

**Pull request merged**